apps/meet-app/backend/modules/calendar/utils.bal

```diff
@@ -0,0 +1,145 @@
+# + creatorEmail - Event creator Email
+# + return - JSON response if successful, else an error
+public isolated function createCalendarEvent(
+        CreateCalendarEventRequest createCalendarEventRequest, string creatorEmail)
+returns CreateCalendarEventResponse|error {
+
+    // WSO2 participants validation
+    string:RegExp wso2EmailDomainRegex = re `^([a-zA-Z0-9_\-\.]+)(@wso2\.com|@ws02\.com)$`;
+    foreach string participant in createCalendarEventRequest.wso2Participants {
+        if (!wso2EmailDomainRegex.isFullMatch(participant.trim())) {
+            return error("Failed to create the calendar event.");
+        }
+    }
+
+    // Replace creator email with a mailto link and add separator before event description.
+    string updatedDisclaimer = replaceCreatorEmail(disclaimerMessage, creatorEmail);
+    string separator = string `<hr style="border: none; border-top: 2px solid #ccc; margin: 15px 0;"><br>`;
+    string updatedDescription = updatedDisclaimer + separator + createCalendarEventRequest.description;
+
+    // Format the event payload as required by the Google Calendar API.
+    CreateCalendarEventPayload calendarEventPayload = {
+        summary: createCalendarEventRequest.title,
+        description: updatedDescription,
+        'start: {
+            dateTime: createCalendarEventRequest.startTime,
+            timeZone: createCalendarEventRequest.timeZone
+        },
+        end: {
+            dateTime: createCalendarEventRequest.endTime,
+            timeZone: createCalendarEventRequest.timeZone
+        },
+        attendees: [
+            ...createCalendarEventRequest.wso2Participants.map((email) => ({email: email.trim()})),
+            ...createCalendarEventRequest.externalParticipants.map((email) => ({email: email.trim()}))
+        ],
+        guestsCanModify: true,
+        conferenceData: {
+            createRequest: {
+                requestId: uuid:createType4AsString(),
+                conferenceSolutionKey: {
+                    'type: "hangoutsMeet"
+                }
+            }
+        }
+    };
+
+    // Create the payload for the API request.
+    http:Request req = new;
+    json calendarEventPayloadJson = calendarEventPayload.toJson();
+    req.setPayload(calendarEventPayloadJson);
+
+    // Make the API call to create the event.
+    http:Response|error response = calendarClient->post(string `/events/${calendarId}?sendUpdates=all`, req);
+    if response is error {
+        return error("Failed to create the calendar event.");
+    }
+
+    // Check if the event was created successfully.
+    if response.statusCode == 201 {
+        json responseJson = check response.getJsonPayload();
+        CreateCalendarEventResponse createCalendarEventResponse = check responseJson
+        .cloneWithType(CreateCalendarEventResponse);
+        return createCalendarEventResponse;
+    }
+
+    return error("Failed to create the calendar event.");
```

Comment:
```suggestion
    http:Response response = check calendarClient->post(string `/events/${calendarId}?sendUpdates=all`, req);

    // Check if the event was created successfully.
    if response.statusCode == 201 {
        json responseJson = check response.getJsonPayload();
        CreateCalendarEventResponse createCalendarEventResponse = check responseJson
        .cloneWithType(CreateCalendarEventResponse);
        return createCalendarEventResponse;
    }

    return error("Failed to create the calendar event.");
```

We need t return the original error to the root level. so we can simply use check. in this approach we don't have access to the original error.